lisa/sut_orchestrator/azure/features.py

```diff
@@ -596,13 +618,97 @@ def get_raw_data_disks(self) -> List[str]:
             disk_array[int(disk.split("/")[-1].replace("lun", ""))] = cmd_result.stdout
         return disk_array
 
+    def add_data_disk(
+        self,
+        count: int,
+        type: schema.DiskType,
+        size_in_gb: int,
+    ) -> List[str]:
+        disk_sku = _disk_type_mapping.get(type, None)
+        assert disk_sku
+        assert self._node.capability.disk
+        assert isinstance(self._node.capability.disk.data_disk_count, int)
+        current_disk_count = self._node.capability.disk.data_disk_count
+        platform: AzurePlatform = self._platform  # type: ignore
+        compute_client = get_compute_client(platform)
 
-__disk_type_mapping: Dict[schema.DiskType, str] = {
-    schema.DiskType.PremiumSSDLRS: "Premium_LRS",
-    schema.DiskType.Ephemeral: "Ephemeral",
-    schema.DiskType.StandardHDDLRS: "Standard_LRS",
-    schema.DiskType.StandardSSDLRS: "StandardSSD_LRS",
-}
+        # create managed disk
+        managed_disks = []
+        for i in range(count):
+            name = f"lisa_data_disk_{i+current_disk_count}"
+            async_disk_update = compute_client.disks.begin_create_or_update(
+                self._resource_group_name,
+                name,
+                {
+                    "location": self._location,
+                    "disk_size_gb": size_in_gb,
+                    "sku": {"name": disk_sku},
+                    "creation_data": {"create_option": DiskCreateOption.empty},
+                },
+            )
+            managed_disks.append(async_disk_update.result())
+
+        # attach managed disk
+        vm = compute_client.virtual_machines.get(
+            self._resource_group_name, self._vm_name
+        )
+        for i, managed_disk in enumerate(managed_disks):
```

Comment:
It's better to validate the existing disk count is the same as capability.disk.data_disk_count. If not, raise an exception.

Reply:
This might not always match. There could be disks added from the provisioning phase.